langchain4j-azure-ai-search/src/main/java/dev/langchain4j/rag/content/retriever/azure/search/AzureAiSearchContentRetriever.java

```diff
@@ -176,9 +176,14 @@ public List<Content> retrieve(Query query) {
 
             List<EmbeddingMatch<TextSegment>> searchResult = super.search(request).matches();
             return searchResult.stream()
-                    .map(EmbeddingMatch::embedded)
-                    .map(Content::from)
-                    .collect(toList());
+                    .map(embeddingMatch -> Content.from(
+                            embeddingMatch.embedded(),
+                            Map.of(
+                                    "score", embeddingMatch.score(),
+                                    "embeddingId", embeddingMatch.embeddingId()
+                            )
+                    ))
+                    .toList();
```

Comment:
Thank you, @emredmrcan 
Would it be possible to add a unit test?

Reply:
Hi @kpavlov, I’ve added a metadata assertion for the integration tests and included a unit test for the case where `AzureAiSearchQueryType` is `VECTOR`, as it doesn’t require `searchClient`. Unfortunately, I couldn’t find a way to mock `searchClient` without refactoring the code to accept pre-configured/mocked clients. 
Let me know if you have any suggestions or additional feedback. Otherwise, please feel free to resolve the conversation.